src/assets/TokensController.ts

```diff
@@ -116,6 +129,33 @@ export class TokensController extends BaseController<
     );
   }
 
+  /**
+   * Fetch metadata for a token.
+   *
+   * @param tokenAddress - The address of the token.
+   * @returns The token metadata.
+   */
+  private async fetchTokenMetadata(
+    tokenAddress: string,
+  ): Promise<TokenListToken | undefined> {
+    try {
+      const token = await fetchTokenMetadata<TokenListToken>(
+        this.config.chainId,
+        tokenAddress,
+        this.abortController.signal,
+      );
+      return token;
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        error.message.includes(TOKEN_METADATA_NO_SUPPORT_ERROR)
+      ) {
+        return undefined;
+      }
+      throw error;
```

Comment:
This path is not covered by a test. The easiest way would be to add a test where the token API returned invalid JSON, or returned a response with `error` set in the response body.